### PR TITLE
Improvements and bug fixes for the TOF Converter window

### LINKED ISSUE #33898: Minor Usability Issues in TOF Coverter
**Describe the bug**
### Angle not Required
- [ ] Converting between Q and d-should not need the scattering angle (Q = 2pi/d)
image
But the value goes negative if 2theta > 360
image
There's no such thing as negative dSpacing...

### Same Unit Conversion
- [ ] It might be nice not to allow the user to convert to the same unit , e.g. TOF -> TOF
In the case you do allow it you shouldn't need the path length!
image

### Single Unit Workspace
- [ ] TOF converter should be able to use a single valued workspace

**To Reproduce**

For example
1. Launch workbench
2. Navigate to Interfaces -> Utility -> TOFConverter
3. Fill in the boxes as shown above

**Platform/Version (please complete the following information):**
 - OS: All
 - OS Version: N/A
 - Mantid Version: 6.3 Babylon Nightly

### LINKED ISSUE #35585: Stop clearing values on every drop-down change in TOFConverter
In the TOFConverter the `Scattering Angle` and `Total flight path` fields are cleared when either the `Units` or `Convert To` drop-downs are changed. We noticed this during manual testing for the release and found that it cleared values that we wanted to keep. We'll need to decide if it's most appropriate to never clear the values, or to only clear them when the fields are disabled (they don't seem to be relevant for all conversions).

### PR DESCRIPTION
**Pull request by jclarkeSTFC**
Some work on the TOF Converter window that's a combination of an improved layout, some refactoring, and some bug fixes. These changes fix a couple of issues (#33898 and #35585)

**Purpose of work**

Fixes #33898 and fixes #35585 

**Summary of work**

- Changed layout of window so the boxes are now centred.
- If the input and output units are the same then the convert button is disabled.
- The inputs for scattering angle and TOF are not cleared when changing units, unless you change to a unit that does not require those values.
- The scattering angle is normalised to [0, 360) degrees before use in the calculation, which prevents oddities like negative d-spacing noted in #33898.
- Disable scattering angle input when converting between d-spacing and momentum transfer, because in this case the angle is not required.
- Option added to take input value from any single-value workspace.

**To test:**

- Check that unit conversions match previous version. Can also check that 1  meV = 1.6 × 10-22 J = 0.24 THz = 8.1 cm-1 = 11.6 K, which is from some notes.
- Try inputting angles outside the range [0, 360).
- Try converting between d-spacing and momentum transfer and check the scatting angle and TOF boxes are disabled.
- Try adding some single-value workspaces, some other workspaces, then check that the input option behaves correctly.

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/TofConverter/convertUnits.py [hunk 1]
```diff
@@ -11,6 +11,12 @@
 
 
 def doConversion(inputval, inOption, outOption, theta, flightpath):
+    if (
+        (inOption == MOMENTUM_TRANSFER or inOption == D_SPACING)
+        and (outOption == MOMENTUM_TRANSFER or outOption == D_SPACING)
+        and inOption != outOption
+    ):
```
Comment (jhaigh0) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/TofConverter/convertUnits.py:18]: I think this is equivalent?

```suggestion
    if (inOption == MOMENTUM_TRANSFER and outOption == D_SPACING)
        or (inOption == D_SPACING and outOption == MOMENTUM_TRANSFER):
```

**Review by jhaigh0 — CHANGES_REQUESTED** [review]
Some initial code changes for you.

**Review by jhaigh0 — APPROVED** [review]
Thanks for the changes 👍

**Comment by gemmaguest** [comment]
@jclarkeSTFC In general I'd prefer see separate PRs rather than one larger PR containing multiple changes like this. Don't be afraid of creating tiny PRs! In general, the smaller the better :) I know that can feel like unnecessary overhead, but I'd rather focus on reducing the overhead of getting PRs through if that is the case! Advantages of smaller PRs include:

- they are (usually) easier to review;
- you will get better feedback;
- they usually get merged more quickly;
- if there are any reviewer changes to address with one change it doesn't hold up the other changes;
- the history of why a change was made is clearer;
- if a problem is introduced it can be tracked down more easily;
- if necessary, one change can be reverted without affecting other changes.

Of course it might make sense to do them together for some reason, e.g. they make sense as a single release note or make life easier for the tester - it's perhaps worth mentioning it in the PR description if there was a deliberate decision along those lines.

**Pull request merged**